bump conda package versions on Appveyor

grlee77 opened the pull request:
Checking the output of:
    ```conda search numpy[subdir=win-64] --info -c defaults | grep main```
    ```conda search scipy[subdir=win-64] --info -c defaults | grep main```
    ```conda search cython[subdir=win-64] --info -c defaults | grep main```

seems to indicate that:
`numpy 0.11.3`
`scipy 0.19.1`
`cython 0.26.1`

is probably the oldest combo that will work for the current defaults channel. I think that we can probably drop support for versions older than these at this point.

TODO: 
- [ ] update the docs accordingly

grlee77 commented:
it looks like the tests on Travis only use a current version of numpy/scipy/cython so I don't think there is a corresponding change to be made there.

hgomersall commented:
This is fine IMO. We'll await the tests to complete and then merge.

peterbell10 commented:
Conda is installing now but the tests are still failing
```
======================================================================
ERROR: test.test_pyfftw_builders (unittest.loader.ModuleImportFailure)
----------------------------------------------------------------------
ImportError: Failed to import test module: test.test_pyfftw_builders
Traceback (most recent call last):
  File "C:\Miniconda\envs\build_env\lib\unittest\loader.py", line 254, in _find_tests
    module = self._get_module_from_name(name)
  File "C:\Miniconda\envs\build_env\lib\unittest\loader.py", line 232, in _get_module_from_name
    __import__(name)
  File "C:\projects\pyfftw\test\test_pyfftw_builders.py", line 35, in <module>
    from pyfftw import builders, empty_aligned, byte_align, FFTW
  File "C:\projects\pyfftw\pyfftw\__init__.py", line 41, in <module>
    from . import config
ImportError: cannot import name config
```

grlee77 commented:
For Python 3.4, a different failure occurs in the import of scipy itself

```
======================================================================
ERROR: test.test_pyfftw_scipy_interface (unittest.loader._FailedTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Miniconda34\envs\build_env\lib\unittest\case.py", line 58, in testPartExecutor
    yield
  File "C:\Miniconda34\envs\build_env\lib\unittest\case.py", line 580, in run
    testMethod()
  File "C:\Miniconda34\envs\build_env\lib\unittest\loader.py", line 33, in testFailure
    raise self._exception
ImportError: Failed to import test module: test.test_pyfftw_scipy_interface
Traceback (most recent call last):
  File "C:\Miniconda34\envs\build_env\lib\unittest\loader.py", line 323, in _find_tests
    module = self._get_module_from_name(name)
  File "C:\Miniconda34\envs\build_env\lib\unittest\loader.py", line 301, in _get_module_from_name
    __import__(name)
  File "C:\projects\pyfftw\test\test_pyfftw_scipy_interface.py", line 35, in <module>
    from pyfftw.interfaces import scipy_fftpack
  File "C:\projects\pyfftw\pyfftw\interfaces\scipy_fftpack.py", line 65, in <module>
    from scipy.fftpack import (dct, idct, dst, idst, diff, tilbert, itilbert,
  File "C:\Miniconda34\envs\build_env\lib\site-packages\scipy\fftpack\__init__.py", line 97, in <module>
    from .basic import *
  File "C:\Miniconda34\envs\build_env\lib\site-packages\scipy\fftpack\basic.py", line 12, in <module>
    from . import _fftpack
ImportError: DLL load failed: The specified module could not be found.
```

There is also a failure related to Dask on 2.7:

```
======================================================================
ERROR: test.test_pyfftw_base (unittest.loader.ModuleImportFailure)
----------------------------------------------------------------------
ImportError: Failed to import test module: test.test_pyfftw_base
Traceback (most recent call last):
  File "C:\Miniconda-x64\envs\build_env\lib\unittest\loader.py", line 254, in _find_tests
    module = self._get_module_from_name(name)
  File "C:\Miniconda-x64\envs\build_env\lib\unittest\loader.py", line 232, in _get_module_from_name
    __import__(name)
  File "C:\projects\pyfftw\test\test_pyfftw_base.py", line 35, in <module>
    from pyfftw import FFTW, _supported_types, _all_types_human_readable
  File "C:\projects\pyfftw\pyfftw\__init__.py", line 43, in <module>
    from . import interfaces
  File "C:\projects\pyfftw\pyfftw\interfaces\__init__.py", line 250, in <module>
    from dask.array.fft import fft_wrap
  File "C:\Miniconda-x64\envs\build_env\lib\site-packages\dask\array\__init__.py", line 5, in <module>
    from .core import (Array, block, concatenate, stack, from_array, store,
  File "C:\Miniconda-x64\envs\build_env\lib\site-packages\dask\array\core.py", line 29, in <module>
    from . import chunk
  File "C:\Miniconda-x64\envs\build_env\lib\site-packages\dask\array\chunk.py", line 73, in <module>
    nancumprod = npcompat.nancumprod
AttributeError: 'module' object has no attribute 'nancumprod'
```

grlee77 commented:
The Dask failure seems to be due to trying to acces NumPy's `nancumprod` which was introduced in NumPy 
1.12

grlee77 commented:
The failure to import config in PyFFTW itself is probably due to some earlier errors in linking:
`fatal error LNK1181: cannot open input file 'libfftw3_omp-3.lib'`
I will have to check what recent Windows builds of FFTW look like and see if there is some change that could have broke things. This might be due to the switch to newer compilers earlier this year.

grlee77 commented:
Dask 0.16 wasn't available on Python 3.4. I am now trying to just switch from 3.4 to 3.7 as that should be more relevant to future releases. Python 3.7 also does not have packages for all of the minimum versions, so I am will test with the most recent dependencies there.

grlee77 commented:
I am going to merge this as-is now that it is finally working again.

@hgomersall:  a couple of changes were made since you approved it
1.) I ended up bumping Scipy to 1.0, but it will probably still work if we were to reset back to 0.19.1.  
2.) I added Python 3.7 cases and 32-bit Python 3.6 and removed 3.4
If you have a problem with either of these I can open a follow-up PR

hgomersall commented:
I guess I broke protocol by approving before tests passing. Oh well, I still approve it - thanks!
